src/dvc_data/index/collect.py

```diff
@@ -101,6 +101,12 @@ def collect(  # noqa: C901, PLR0912
                 fsid = data.fs.fsid
             except (NotImplementedError, AttributeError):
                 fsid = data.fs.protocol
+            except BaseException as exc:  # noqa: BLE001
```

Comment:
Even if we update dvcfs to re-raise the SCMError as an AttributeError instead when loading `fs.fsid`, this just pushes the problem further down the line. In this case we will still hit the same SCMError when trying to clone the repo again upon the next fs function call.

I'm not sure if catching the broad exception here is right though, we could re-raise ValueError in dvcfs (and then catch that here instead of BaseException)

cc @efiop

Reply:
Yeah, `BaseException` indeed feels a bit too broad.

> In this case we will still hit the same SCMError when trying to clone the repo again upon the next fs function call.

This might be fine, as we try to connect to check for auth errors anyway. Not sure if AttirubteError specifically is ideal though, maybe it is. Or `BaseException` might actually be alright to handle here like you did already, since it is about whatever fsid might throw at us... So sounds like I reach your coclusion as well here 😄

Reply:
Should we still collect by falling back to `data.fs.protocol`?

Reply:
In the general case it would probably be better to have fallback behavior, but for dvcfs/gitfs falling back here will just delay the failure to the next gitfs call (when it will retry cloning the repo and fail)